### PR TITLE
[FIX] website_editor: use background image on the right element

### PR DESCRIPTION
**Pull request by robinlej**
[FIX] website_editor: use background image on the right element

Commit [1] addressed the following situation: when we have a parallax on
an element, it adds the background on a `<span>` child of said element
instead of applying it directly on the element. Retargeting the `<span>`
is needed to display all the options linked to the background image.

Following [1], building blocks whose children are allowed to have a
background image are also applied their 1st child's background when one
of their children's background is modified. This happens because, when
changing the background of a child item, `BackgroundOptimize` is called
both on the child and on the main snippet due to
`_onBackgroundChanged()` bubbling up and `_loadImageInfo()` targetting
any child in the tree with the class `.oe_img_bg`.

This commit prevents `_onBackgroundChanged()` from bubbling up and
modifies the behavior from [1] by making sure the element identified as
the target within `_loadImageInfo()` can only be either the original
target or the parallax element.

Steps to reproduce:
- Drop a carousel building block
- Change its style to bordered
- Change the background image of the 2nd slide
=> the background of the entire snippet is also changed. It is then
impossible to remove the image on the snippet.

task-3349669

[1]: https://github.com/odoo/odoo/commit/31ba906

### CODE REVIEW

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

**Comment by xO-Tx** [comment]
@robinlej Sorry for the late reply about this! I had to check the code myself to be sure about what exactly happens... I see that we have three mixed issues here:

**A-** The `_onBackgroundChanged()` > `_autoOptimizeImage` is called twice on different targets (carousel & carousel item) when we change item's background image (as discussed last time on discord...):

For this one, I think it's caused by the `"background_changed"` event that bubbles up when the background is updated on a carousel item leading to the main issue you specified in the commit message, so adding a simple `stopPropagation();` to `_onBackgroundChanged()` should solve the problem (TO TEST...).

**B-** The `_loadImageInfo()` will target a child element's background if no background image was found on the parent:

You can clearly see this even before changing the background image (The carousel will show `BackgroundOptimize` options without any background image added...)


https://github.com/odoo/odoo/assets/25032876/388ad9aa-efb2-4b2c-ad3c-3d199612f246


This is caused by the code in [this diff](https://github.com/odoo/odoo/commit/31ba906#diff-ffb61e86e6b8297ef8997f9c605de14de896b54161c032d88a80bbaeac4f89beR3635-R3636) that looks in the DOM for a new target when a background image is not set... Normally, if no background image was found on `this.$target`, the code should simply consider that there is no background set... But there is an exception to this! which leads us to the third issue (also the issue that https://github.com/odoo/odoo/commit/31ba906 was added to fix):

**C-** The `_loadImageInfo()` doesn't take in consideration the target update caused by "Parallax":

As you can see [here](https://github.com/odoo/odoo/blob/14.0/addons/website/static/src/js/editor/snippets.options.js#L1867-L1873), the parallax option does the job correctly by sending the new parallax target to background options... But on `BackgroundOptimize`, the `_loadImageInfo()` will set `this.img` using the wrong target before the `notify()` method could change its `$target` to the correct one. I think there is many options to fix this issue... one of them is to adapt the target as in the current diff of your commit (maybe a `trigger("background_changed");` at the end of `notify()` works too...) but I let you choose the best implementation, just don't forget to specify in your comment that Parallax sends the right target but `_loadImageInfo` should wait for the update in `notify()` to work correctly...

So to sum up, we should prevent the `"background_changed"` event propagation for **[A]** and use the right target on `_loadImageInfo()` as mentioned in **[B]** and **[C]** :+1:

**Review by xO-Tx — APPROVED** [review]
@robinlej Thanks for the updates :+1:

@robodoo r+

**Comment by xO-Tx** [comment]
@robodoo retry
